lib/ansible/plugins/cache/__init__.py

```diff
@@ -320,10 +320,7 @@ def __len__(self):
     def _do_load_key(self, key):
         load = False
         if key not in self._cache and key not in self._retrieved and self._plugin_name != 'memory':
-            if isinstance(self._plugin, BaseFileCacheModule):
-                load = True
-            elif not isinstance(self._plugin, BaseFileCacheModule) and self._plugin.contains(key):
-                # Database-backed caches don't raise KeyError for expired keys, so only load if the key is valid by checking contains()
+            if self._plugin.contains(key):
```

Comment:
at this point, a single `if` is needed (maby use 'any' to make it more readable?)